msdk-featuredetection-adap3d/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -160,15 +165,15 @@ public ADAP3DFeatureDetectionMethod(@Nonnull RawDataFile rawFile, @Nullable Pred
         Collections.max(peakWidthList) + LOW_BOUND_PEAK_WIDTH_PERCENT * avgPeakWidth;
 
     // set the estimated parameters.
-    objParameters.setLargeScaleIn(highestWaveletScale);
-    objParameters.setMinPeakWidth(peakDurationLowerBound);
-    objParameters.setMaxPeakWidth(peakDurationUpperBound);
-    objParameters.setCoefAreaRatioTolerance(coefOverAreaThreshold);
+    parameters.setLargeScaleIn(highestWaveletScale);
```

Comment:
I see a problem here that the parameter values are always overwritten with the estimated ones - even if the user provided pre-defined parameters in the constructor. It should probably be implemented in a different way - the default value for all the parameters should be null, and user can set some of them. Those that are not provided by the user should be set during the estimation stage.

Reply:
Our idea was that user would set up initial parameters (or use the default values), and than those parameters are estimated from the first detected peaks. But I see your point. I need to discuss it to Xiuxia.